gosrc/github.go

```diff
@@ -101,14 +102,24 @@ func getGitHubDir(client *http.Client, match map[string]string, savedEtag string
 		return nil, ErrNotModified
 	}
 
+	p, err := c.getBytes(expand("https://api.github.com/repos/{owner}/{repo}/contents{dir}?ref={tag}", match))
+	if err != nil {
+		return nil, err
+	}
+	p = bytes.TrimSpace(p)
+
+	// GitHb returns JSON array for directories
+	if len(p) == 0 || p[0] != '[' {
+		return nil, NotFoundError{Message: "Not a directory in GitHub"}
+	}
+
```

Comment:
Could Go's `encoding/json` package be used to detect this situation, instead of looking at the raw bytes? In other words, can this be done at a higher level?

For example. Try to do what was being done before. If there's an error, check if the error is caused by this being a directory. If so, return `NotFoundError`.